test/unit/callbacks/afterUpdate.update.js

```diff
@@ -0,0 +1,78 @@
+var Waterline = require('../../../lib/waterline'),
+  assert = require('assert');
+  _ = require('lodash');
+
+describe('.afterUpdate()', function() {
+  var User, afterUpdateValues = null;
+
+  before(function (done) {
+    var waterline = new Waterline();
+    var Model = Waterline.Collection.extend({
+      identity: 'user',
+      connection: 'foo',
+      attributes: {
+        name: 'string',
+        employed: 'boolean',
+        meta: 'json',
+        hobbies: 'array'
+      },
+
+      afterUpdate: function (values, cb) {
+        afterUpdateValues = _.cloneDeep(values);
+        cb();
+      }
+    });
+
+    waterline.loadCollection(Model);
+
+    // Fixture Adapter Def
+    var adapterDef = {
+      update: function (con, col, criteria, values, cb) {
+        // emulate values coming back from (for example) sails-mysql adapter
```

Comment:
I think this is MySQL specific, so this test should be in the waterline-adapter-tests.